packages/libs/deer/desert/src/tape.rs

```diff
@@ -0,0 +1,166 @@
+use core::{
+    ops::{Deref, Range},
+    slice::SliceIndex,
+};
+
+use bitvec::{
+    boxed::BitBox,
+    order::Lsb0,
+    prelude::{BitSlice, BitVec},
+    slice::BitSliceIndex,
+};
+
+use crate::token::Token;
+
+#[derive(Debug)]
+enum Trivia<'a> {
```

Comment:
May I ask what the origin of this naming is? Seems interesting :cow:

Reply:
Yes for sure! This is taken from `rust-analyser`, where whitespace and comments are regarded as trivia. The idea in this case is basically: 

```json
{"key1": 1, "key2": 2, "key3": 3}
```

Taking this, the whitespace would be considered trivia, when we then search for a key, the key and value span are both "erased" and converted to trivia, e.g. taking `key2`, our object internally turns into:

```json
{"key1": 1,            "key3": 3}
```


https://github.com/rust-lang/rust-analyzer/blob/80cabf726068187d8686b5ccf37aac484da84904/crates/parser/src/syntax_kind.rs#L24-L29